### PR TITLE
DTSPO-70 Add CI Checks

### PR DESCRIPTION
**Pull request by zc-hmcts**
### JIRA link  ###
https://tools.hmcts.net/jira/browse/DTSPO-70

### Change description ###
Adds CI checks for pull requests

**Does this PR introduce a breaking change?**

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by adusumillipraveen — APPROVED** [review]

File: tests/test-kustomize.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+set -x
+
+curl -s "https://raw.githubusercontent.com/\
+kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" -o install_kustomize.sh && chmod +x install_kustomize.sh && ./install_kustomize.sh 3.7.0
+
+
+kustomizepaths=(
+    k8s/environments/demo/cluster-00-overlay
+    k8s/environments/demo/cluster-01-overlay
+    # k8s/environments/demo/common-overlay
```
Comment (timja) [inline tests/test-kustomize.sh:11]: why are these commented out?

File: tests/test-kustomize.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# for filepath in "${kustomizepaths[@]}"; do
+#     ./kustomize build --load_restrictor none "$filepath" >/dev/null
+#     if [ $? -eq 1 ]
+#     then
+#      echo "Kustomize failing for env $filepath" && exit 1
+#     fi
+# done
+
+prod_whitelist_helm_release_pattern='toffee\|video-hearings' # Helm Release names seperated by `\|`
```
Comment (timja) [inline tests/test-kustomize.sh:40]: why are these 'whitelisted' which I think means to allow disabling flux automation in prod?

**Review by timja — CHANGES_REQUESTED** [review]
2 comments

File: tests/test-kustomize.sh (hunk 1 above)
Comment (zc-hmcts) [reply tests/test-kustomize.sh]: uncommented

File: tests/test-kustomize.sh (hunk 2 above)
Comment (zc-hmcts) [reply tests/test-kustomize.sh]: `grep -v "${!env_white_list}"` is a double negative

File: tests/test-kustomize.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+for filepath in "${kustomizepaths[@]}"; do
+     ./kustomize build --load_restrictor none "$filepath" >/dev/null
+     if [ $? -eq 1 ]
+     then
+      echo "Kustomize failing for env $filepath" && exit 1
+     fi
+done
+
+prod_whitelist_helm_release_pattern='vh\|toffee' # Helm Release names seperated by `\|`
```
Comment (timja) [inline tests/test-kustomize.sh:40]: why are vh and toffee excluded here?

Comment (zc-hmcts) [reply tests/test-kustomize.sh]: removed from exclusion, replaced with "sample" since it doesn't work when empty

**Comment by timja** [comment]
you have conflicts that need resolving

File: k8s/environments/prod/cluster-00/mi/mi-data-extractor.yaml [hunk 4]
```diff
@@ -25,7 +25,7 @@ spec:
       memoryRequests: '2048Mi'
       cpuRequests: '200m'
       memoryLimits: '4096Mi'
-      activeDeadlineSeconds: 82800 #23 hours
+      activeDeadlineSeconds: 82800#23 hours
```
Comment (msl8r) [inline k8s/environments/prod/cluster-00/mi/mi-data-extractor.yaml:28]: @zc-hmcts

File: k8s/environments/prod/cluster-00/mi/mi-integration-service.yaml [hunk 5]
```diff
@@ -22,7 +22,7 @@ spec:
       suspend: false
       kind: CronJob
       schedule: '15 1 * * *'
-      activeDeadlineSeconds: 82800 #23 hours
+      activeDeadlineSeconds: 82800#23 hours
```
Comment (msl8r) [inline k8s/environments/prod/cluster-00/mi/mi-integration-service.yaml:25]: @zc-hmcts

**Pull request merged**